src/frontend/src/pages/AdminToolsPage/AdminToolsUserManagement.tsx

```diff
@@ -0,0 +1,142 @@
+    if (value) {
+      const user = data.find((user) => fullNamePipe(user) === value.split(' -')[0]);
+      if (user) setUser(user);
+    } else {
+      setUser(null);
+    }
+  };
+
+  const handleRoleChange = (event: SelectChangeEvent) => {
+    setRole(event.target.value);
+    if (user && event.target.value === user.role) {
+      setIsDisabled(true);
+    } else {
+      setIsDisabled(false);
+    }
+  };
+
+  const handleClick = async () => {
+    setHideSuccessLabel(true);
+    await update.mutateAsync({ userId: user?.userId, role }).catch((error) => {
+      alert(error);
+      throw new Error(error);
+    });
+    setHideSuccessLabel(false);
+  };
+
+  return (
+    <PageBlock title={'Role Management'}>
+      <Grid container spacing={2}>
+        <Grid item xs={12} md={8}>
+          <Autocomplete
+            disablePortal
+            id="autocomplete"
+            onChange={handleSearchChange}
+            options={data.map((user) => `${fullNamePipe(user)} - ${user.email}`)}
+            sx={{
+              height: '40px',
+              backgroundColor: 'ActiveBorder',
+              width: '100%',
+              borderRadius: '25px',
+              borderColor: 'black',
+              '.MuiOutlinedInput-notchedOutline': {
+                border: 0,
+                borderRadius: '25px'
+              }
+            }}
+            size="small"
+            renderInput={(params) => (
+              <TextField
+                {...params}
+                InputProps={{
+                  ...params.InputProps,
+                  startAdornment: (
+                    <InputAdornment position="start">
+                      <SearchIcon />
+                    </InputAdornment>
+                  )
+                }}
+                placeholder="Select a User"
+              />
+            )}
+          />
+        </Grid>
+        <Grid item xs={12} md={4}>
+          <FormControl
+            size="small"
+            sx={{
+              width: '100%',
+              backgroundColor: '#ff0000',
+              borderColor: '#0062cc',
+              borderRadius: '25px',
+              height: '40px'
+            }}
+          >
+            <Select
+              displayEmpty={true}
+              renderValue={(value) => (value ? value : user ? user.role : 'Current Role')}
+              id="role-select"
+              value={role}
+              onChange={handleRoleChange}
+              disabled={!user}
+              sx={{
+                '.MuiOutlinedInput-notchedOutline': { border: 0, borderRadius: '25px', borderColor: 'black' }
+              }}
+            >
+              <MenuItem value={'ADMIN'}>Admin</MenuItem>
+              <MenuItem value={'LEADERSHIP'}>Leadership</MenuItem>
+              <MenuItem value={'MEMBER'}>Member</MenuItem>
+              <MenuItem value={'GUEST'}>Guest</MenuItem>
+            </Select>
+          </FormControl>
```

Comment:
I think this select is widely usable enough that it's worth moving into another file in the components folder. Probably gonna make a ticket for abstracting it out into a component after this ticket is done